scripts/run-registry.ts

```diff
@@ -146,25 +136,31 @@ const run = async () => {
 
   logger.log(`🌿 verdaccio running on ${verdaccioUrl}`);
 
-  // in some environments you need to add a dummy user. always try to add & catch on failure
-  try {
-    await addUser(verdaccioUrl);
-  } catch (e) {
-    //
-  }
+  logger.log(`👤 add temp user to verdaccio`);
+  await execa(
+    'npx',
+    // creates a .npmrc file in the root directory of the project
+    ['npm-auth-to-token', '-u', 'foo', '-p', 's3cret', '-e', 'test@test.com', '-r', verdaccioUrl],
+    {
+      cwd: root,
+    }
+  );
```

Comment:
This new code is different than before.

The old method was wrapped in a try catch, and so even if the call to create a user failed, the publosh would continue. Now it break the script.

Reply:
Your observation is correct. If the user didn’t have a .npmrc file set with an authToken in place from a previous no-link run, publishing packages simply didn't work recently.Therefore for us core maintainers it was never occurring, because everyone of us had a npmrc file in `~` already in place with some authToken information. A user has to be authenticated, because otherwise publishing just does not work. You can revert that change locally and delete your global npmrc file and you will stumble upon an error were Verdaccio complains that the user is not authenticated. Wrapping the code in a try-catch block with a noop therefore doesn’t make any sense. It seems that either a newer npm or Verdaccio version is enforcing authentication.

Reply:
Hmm, I still don't quite fully understand the issue, but I'll trust that this is the right thing to do then.

Reply:
I try to explain it with other words:

- it seems that Verdaccio OR npm (potentially introduced with a newer version) enforces authentication, because otherwise publishing packages isn't allowed
- Therefore the try catch block is obsolete, because if adding/logging-in a user does not work, the publishing afterwards will also not work. 
- The package `npm-cli-adduser` doesn't seem to work anymore, at least locally. I encountered the issue by deleting the npmrc file in my user directory (`~`). Running the publishing with Verdaccio started then to fail because the script adding the user failed and therefore the publishing afterwards. I also tried the documented legacy mode, but it was also failing. Ironically, I had the same issue on my bun branch, but I thought it is related to the bun runtime.
- The new CLI (which is also used in Verdaccio's docker image) will create a temporary npmrc in the Storybook root folder instead, which is also a better approach, because it does not pollute the user's global config.

I hope that makes more sense now!